sdk/table/azure-table/azure/table/_shared/table_client_base.py

```diff
@@ -0,0 +1,60 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+
+from azure.table._shared._error import _validate_table_name
+from azure.table._shared.base_client import parse_query
+from .base_client import StorageAccountHostsMixin
+
+class TableClientBase(StorageAccountHostsMixin):
+    def __init__(
+        self, account_url, # type: str
+        table_name, # type: str
+        credential=None, # type: Union[str,TokenCredential]
+        **kwargs # type: Any
+    ):
+        # type: (...) -> None
+        """Create TableClientBase from a Credential.
+
+        :param account_url:
+            A url to an Azure Storage account.
+        :type account_url: str
+        :param table_name: The table name.
+        :type table_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+        :type credential: Union[str,TokenCredential]
```

Comment:
I know this isn't the point of this PR, but could we remove the references to TokenCredential and the sentence "or an instance of a TokenCredentials class from azure.identity.". Tables does not support auth via AAD quite yet.